MobileCenterDistribute/MobileCenterDistribute/Internals/Version/MSSemVerPreReleaseId.m

```diff
@@ -0,0 +1,83 @@
+#import "MSSemVerPreReleaseId.h"
+
+@implementation MSSemVerPreReleaseId
+
++ (instancetype)identifierWithString:(NSString *)identifier {
+  return [[self alloc] initWithString:identifier];
+}
+
+- (instancetype)initWithString:(NSString *)identifier {
+
+  // Validate.
+  if (!(identifier.length > 0)) {
+    return nil;
+  }
+
+  // Initialize.
+  if ((self = [super init])) {
+    _identifier = identifier;
+  }
+  return self;
+}
+
+- (NSComparisonResult)compare:(MSSemVerPreReleaseId *)identifier {
+  NSString *identifierA = self.identifier;
+  NSString *identifierB = identifier.identifier;
+  NSNumber *idANumValue;
+  NSNumber *idBNumValue;
+  NSUInteger idALength;
+  NSUInteger idBLength;
+
+  // Strictly identical, return it now.
+  if ([identifierA isEqualToString:identifierB]) {
+    return NSOrderedSame;
+  }
+
+  // Compare by numeric values.
+  idANumValue = [self numberValue];
+  idBNumValue = [identifier numberValue];
+  if (idANumValue && idBNumValue) {
+    return [idANumValue compare:idBNumValue];
+  }
+
+  // A is a number but not B.
+  if (idANumValue && !idBNumValue) {
+    return NSOrderedAscending;
+  }
+
+  // B is a number but not A.
+  if (!idANumValue && idBNumValue) {
+    return NSOrderedDescending;
+  }
+
+  // None is numeric only, compare by ASCII order.
+  idALength = identifierA.length;
+  idBLength = identifierB.length;
+  const char *idAASCIIChars = [identifierA cStringUsingEncoding:NSASCIIStringEncoding];
+  const char *idBASCIIChars = [identifierB cStringUsingEncoding:NSASCIIStringEncoding];
+  for (NSUInteger i = 0; i < MIN(idALength, idBLength); i++) {
+    if (*(idAASCIIChars + i) > *(idBASCIIChars + i)) {
+      return NSOrderedDescending;
+    } else if (*(idAASCIIChars + i) < *(idBASCIIChars + i)) {
+      return NSOrderedAscending;
+    }
+  }
+
+  /*
```

Comment:
I think we decided to not start with `/**` for in block comments or I misread our coding standards. I kinda like it this way because with `/*` it draw less attentions over standard `//` than `/**`.